src/contracts/core/AllocationManager.sol

```diff
@@ -292,40 +302,43 @@ contract AllocationManager is
     }
 
     /// @inheritdoc IAllocationManager
-    function setAVSRegistrar(address avs, IAVSRegistrar registrar) external checkCanCall(avs) {
-        _avsRegistrar[avs] = registrar;
-        emit AVSRegistrarSet(avs, getAVSRegistrar(avs));
+    function registerAsAVS(
+        IAVSRegistrar registrar,
+        string calldata metadataURI,
+        CreateSetParams[] calldata params
+    ) external {
+        _updateAVSRegistrar(msg.sender, registrar);
+        _updateAVSMetadataURI(msg.sender, metadataURI);
+        _createOperatorSets(msg.sender, params);
     }
```

Comment:
So I think part of the appeal of this change would be to make "AVS" an "Operator" a unique distinction. As in, you can be one or the other, but not both. ... So I'd recommend adding a check that `msg.sender` is NOT an operator in the core contracts.

That said -- we would need to check whether any existing AVSs are also operators on mainnet, because we don't want this to brick anyone...

Reply:
... right. This is a lot of effort tbh :sweat: